solutions/usaco-994.mdx

```diff
@@ -2,49 +2,133 @@
 id: usaco-994
 source: USACO Gold 2020 January
 title: Farmer John Solves 3SUM
-author: Qi Wang
+author: Qi Wang, Ryan Chou
 ---
 
 [Official Analysis (C++ and Java)](http://www.usaco.org/current/data/sol_threesum_gold_jan20.html)
 
+## Explanation
+
+Although the problem may be daunting at first, it helps to think about how a smaller range is connected to another.
+
+If $\texttt{ways}[i][j]$ equals the number of unordered triplets which sum to zero between $[i, j]$, then what states does this rely on?
+
+![](usaco-994/init.png)
+
+We can choose to exclude one endpoint or the other, which gives us a picture like this (reminder to make sure not to double count the overlap).
+
+![](usaco-994/ps.png)
+
+While this'll count all the triplets in between them, we aren't taking into account any triplets which use both the elements at $i$ and $j$, since those aren't included in our "PIE-like" transition.<Asterisk>Currently, our state $(i, j)$ = (triplets which begin at $i$ and end at $i + 2$ to $j - 1$) $+$ (triplets which begin at $i + 1$ and end at $i + 3$ to $j$)</Asterisk>
```

Comment:
i feel like this asterisk only makes things more confusing
"wha"
"what if the interval only has like 3 elements? the intervals in the asterisk would be out of bounds!"